Good start and thank you for your work, but I have reviewed the code and tried to do tests and encountered several issues that require fixing:

**Docker compose pulling non-existing image:**
docker-compose is trying to pull a non-existing image and errors out as there's no CI/CD in place to publish docker images onto Docker Hub:
```
% docker compose up
[+] Running 0/1
 ⠿ ehb Error                                                               2.2s
Error response from daemon: pull access denied for repository/every-hour-bot, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

As we're trying to build it locally, `image: repository/every-hour-bot` should be replaced with `build .`

**Invalid mount config:**
Docker-compose has invalid mount config and will cause Docker to error out. We need to use an absolute path inside of the container
Changing `./media:./media` to `./media:/ehb/media` fixes the issue


**Container trying to use pm2, but its not installed:**
Container doesn't run as Dockerfile doesn't install PM2 and is not included in the Alpine image:
```
Attaching to everyhourbot-ehb-1
everyhourbot-ehb-1  | node:internal/modules/cjs/loader:998
everyhourbot-ehb-1  |   throw err;
everyhourbot-ehb-1  |   ^
everyhourbot-ehb-1  |
everyhourbot-ehb-1  | Error: Cannot find module '/ehb/pm2 start -f index.js --name ${ACCOUNT_NAME}'
everyhourbot-ehb-1  |     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
everyhourbot-ehb-1  |     at Function.Module._load (node:internal/modules/cjs/loader:841:27)
everyhourbot-ehb-1  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
everyhourbot-ehb-1  |     at node:internal/main/run_main_module:22:47 {
everyhourbot-ehb-1  |   code: 'MODULE_NOT_FOUND',
everyhourbot-ehb-1  |   requireStack: []
everyhourbot-ehb-1  | }
everyhourbot-ehb-1 exited with code 1
```

PM2 is recommended for daemonising NodeJS processes in non-container context, but is not required for containers. We can just do `node index.js` as the container will shut down if the process stops.

Fixing the issues will get the container up and running:
```
% docker compose up
[+] Running 1/0
 ⠿ Container everyhourbot-ehb-1  Created                                                                         0.0s
Attaching to everyhourbot-ehb-1
everyhourbot-ehb-1  | Welcome to EveryHourBot Bot.
everyhourbot-ehb-1  | Version 1.1.0
```

Can you please fix the listed issues above and get back to me once it's done? Thank you

--- a/Dockerfile
+++ b/Dockerfile
@@ -11,4 +11,4 @@ RUN chmod 705 /ehb
 RUN npm install
 
 USER daemon
-CMD ["pm2 start -f index.js --name ${ACCOUNT_NAME}"]
+CMD ["node", "index.js"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -6,8 +6,8 @@ RUN mkdir /ehb/media
 WORKDIR /ehb
 COPY . /ehb/
 
-RUN chown daemon /ehb
-RUN chmod 705 /ehb
+RUN chown -R daemon /ehb
+RUN chmod -R 705 /ehb
 RUN npm install
 
 USER daemon

--- a/docker-compose.yaml
+++ b/docker-compose.yaml
@@ -12,4 +12,4 @@ services:
     - MASTODON_SERVER=""
     - MASTODON_ACCESS_TOKEN=""
     volumes:
-    - ./media:./ehb/media 
+    - ./media:/ehb/media 

--- a/docker-compose.yaml
+++ b/docker-compose.yaml
@@ -1,9 +1,8 @@
 version: "3.4"
 services:
   ehb:
-    image: repository/every-hour-bot
+    build: .
     environment:
-    - ACCOUNT_NAME=""
     - CONSUMER_KEY=""
     - CONSUMER_SECRET=""
     - ACCESS_TOKEN_KEY=""
@@ -13,4 +12,4 @@ services:
     - MASTODON_SERVER=""
     - MASTODON_ACCESS_TOKEN=""
     volumes:
-    - ./media:./media 
+    - ./media:./ehb/media 

